ballerina/sql_client.bal

```diff
@@ -24,40 +24,58 @@ public client class SQLClient {
+                         map<JoinMetadata> joinMetadata = {}) returns error? {
         self.entityName = entityName;
         self.tableName = tableName;
         self.fieldMetadata = fieldMetadata;
         self.keyFields = keyFields;
         self.dbClient = dbClient;
+        self.joinMetadata = joinMetadata;
     }
 
     public function runInsertQuery(record {} 'object) returns sql:ExecutionResult|error {
         sql:ParameterizedQuery query = sql:queryConcat(
             `INSERT INTO `, self.tableName, ` (`,
-            self.getColumnNames(true), ` ) `,
+            self.getInsertColumnNames(true), ` ) `,
             `VALUES `, self.getInsertQueryParams('object)
         );
         return check self.dbClient->execute(query);
     }
 
-    public function runReadByKeyQuery(typedesc<record {}> t, anydata... keys) returns record {}|error {
+    public function runReadByKeyQuery(typedesc<record {}> t, anydata key, string[] include = []) returns record {}|error {
         sql:ParameterizedQuery query = sql:queryConcat(
-            `SELECT `, self.getColumnNames(), ` FROM `, self.tableName, ` WHERE `, check self.getGetKeyWhereClauses(keys)
+            `SELECT `, self.getSelectColumnNames(include), ` FROM `, self.tableName, ` AS `, stringToParameterizedQuery(self.entityName)
         );
+
+        string[] joinKeys = self.joinMetadata.keys();
+        foreach string joinKey in joinKeys {
+            if include.indexOf(joinKey) != () {
+                JoinMetadata joinMetadata = self.joinMetadata.get(joinKey);
+                query = sql:queryConcat(query, ` LEFT JOIN `, stringToParameterizedQuery(joinMetadata.refTable + " " + joinKey), ` ON `, stringToParameterizedQuery(joinKey + "." + joinMetadata.refFields[0]), ` = `, stringToParameterizedQuery(self.entityName + "." + (<string[]>joinMetadata.joinColumns)[0]));
+            }
+        }
+
+        query = sql:queryConcat(query, ` WHERE `, check self.getGetKeyWhereClauses(key));
         record {}|error result = self.dbClient->queryRow(query, t);
         if result is sql:NoRowsError {
-            if keys.length() > 1 {
-                return <InvalidKey>error("A record does not exist for '" + self.entityName + "' for key " + keys.toBalString() + ".");
-            }
-            return <InvalidKey>error("A record does not exist for '" + self.entityName + "' for key " + keys[0].toBalString() + ".");
+            return <InvalidKey>error("A record does not exist for '" + self.entityName + "' for key " + key.toBalString() + ".");
         }
         return result;
     }
 
-    public function runReadQuery(typedesc<record {}> t, map<anydata>? filter) returns stream<record {}, sql:Error?>|error {
-        sql:ParameterizedQuery query = sql:queryConcat(`SELECT `, self.getColumnNames(), ` FROM `, self.tableName);
+    public function runReadQuery(typedesc<record {}> t, map<anydata>? filter, string[] include = []) returns stream<record {}, sql:Error?>|error {
```

Comment:
```suggestion
    public function runReadQuery(typedesc<record {}> returnType, map<anydata>? filter, string[] include = []) returns stream<record {}, sql:Error?>|error {
```

Reply:
Shall we use possible names for the input params instead of `t`? please check other places.

Reply:
@niveathika has sent a separate PR with this change
https://github.com/ballerina-platform/module-ballerina-persist/pull/14/files#diff-60e6b5182baa2d4305ac55d2f49fb4e9365319f6fd2d4b3e7b73f09aec711059R46

Shall we merge that PR to do this change?

Reply:
Yes,merged